src/core/Akka.Persistence.Query.Tests/Utils/TimeUuid.cs

```diff
@@ -0,0 +1,270 @@
+﻿//
+//      Copyright (C) DataStax Inc.
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+using System;
+
+namespace Akka.Persistence.Query
+{
+    /// <summary>
+    /// Represents a v1 uuid 
+    /// </summary>
+    internal struct TimeUuid : IEquatable<TimeUuid>, IComparable<TimeUuid>
+    {
```

Comment:
`TimeUuid` implementation has been pulled into the code to enable proper testing.